appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -636,6 +681,31 @@
                ;; If it's not the REPL the form's $define() method will do the registration
                (add-to-events 'component-name 'event-name)))))))
 
+(define (define-generic-event component-name event-name args . body)
+  (let ((event-func-name (gen-event-name #`component-name #`event-name)))
+    (define-event-helper event-func-name args body)
+    (com.google.appinventor.components.runtime.EventDispatcher:registerEventForDelegation
+                (as com.google.appinventor.components.runtime.HandlesEventDispatching *this-form*)
+                'component-name
+                'event-name)))
+
```

Comment:
Sorry, I meant here.

@mark-friedman Hey have you gotten a chance to look this over? I'll be working on it for a good amount of time later today and would really appreciate some feedback/help. Thanks!